recipes/boa-ide

```diff
@@ -0,0 +1,4 @@
+(boa-ide
+ :fetcher github
+ :repo "boalang/syntax-highlight"
+ :files ("boa-sc.el" "boa-ide.el" "boa-study-config.el"))
```

Comment:
For this package, because it's named `boa-ide`, the prefixes have to be aligned with the package name --

`boa-sc.el` with package-lint 20230806.1331:
```
35 issues found:
5:0: error: Package-Requires outside the main file have no effect.
34:0: error: "boa-sc-configuration" doesn't start with package's prefix "boa-ide".
54:0: error: "boa-sc-query" doesn't start with package's prefix "boa-ide".
64:0: error: "boa-sc-processor" doesn't start with package's prefix "boa-ide".
71:0: error: "boa-sc-analysis" doesn't start with package's prefix "boa-ide".
77:0: error: "boa-sc-substitution" doesn't start with package's prefix "boa-ide".
83:0: error: "boa-sc-csv" doesn't start with package's prefix "boa-ide".
94:0: error: "boa-sc-data" doesn't start with package's prefix "boa-ide".
97:0: error: "boa-sc-verbosity" doesn't start with package's prefix "boa-ide".
103:0: error: "boa-sc-maphash" doesn't start with package's prefix "boa-ide".
108:0: error: "boa-sc-convert-analysis" doesn't start with package's prefix "boa-ide".
114:0: error: "boa-sc-convert-substitution" doesn't start with package's prefix "boa-ide".
122:0: error: "boa-sc-convert-csv" doesn't start with package's prefix "boa-ide".
132:0: error: "boa-sc-convert-processor" doesn't start with package's prefix "boa-ide".
143:0: error: "boa-sc-convert-query" doesn't start with package's prefix "boa-ide".
158:0: error: "boa-sc-convert-configuration" doesn't start with package's prefix "boa-ide".
179:0: error: "boa-sc-parse" doesn't start with package's prefix "boa-ide".
210:0: error: "boa-sc-parse-needed-p" doesn't start with package's prefix "boa-ide".
219:0: error: "boa-sc-get-study-config-file" doesn't start with package's prefix "boa-ide".
223:0: error: "boa-sc-get-study-config-buffer" doesn't start with package's prefix "boa-ide".
233:0: error: "boa-sc-get-data" doesn't start with package's prefix "boa-ide".
239:0: error: "boa-sc-get-project-dir" doesn't start with package's prefix "boa-ide".
248:0: error: "boa-sc-get-or-read-project" doesn't start with package's prefix "boa-ide".
262:0: error: "boa-sc-datasets" doesn't start with package's prefix "boa-ide".
270:0: error: "boa-sc-outputs" doesn't start with package's prefix "boa-ide".
282:0: error: "boa-sc-csvs" doesn't start with package's prefix "boa-ide".
295:0: error: "boa-sc-analyses" doesn't start with package's prefix "boa-ide".
303:0: error: "boa-sc-snippets" doesn't start with package's prefix "boa-ide".
311:0: error: "boa-sc-outputs-query" doesn't start with package's prefix "boa-ide".
324:0: error: "boa-sc-csv-query" doesn't start with package's prefix "boa-ide".
338:0: error: "boa-sc-analyses-query" doesn't start with package's prefix "boa-ide".
353:0: error: "boa-sc-snippets-query" doesn't start with package's prefix "boa-ide".
367:0: error: "boa-sc-set-verbose" doesn't start with package's prefix "boa-ide".
386:0: error: "boa-sc-get-verbosity" doesn't start with package's prefix "boa-ide".
403:0: error: "boa-sc-compile" doesn't start with package's prefix "boa-ide".
```

`boa-study-config.el` with package-lint 20230806.1331:
```
14 issues found:
5:0: error: Package-Requires outside the main file have no effect.
40:0: error: "boa-study-config-project-dir" doesn't start with package's prefix "boa-ide".
46:0: error: "boa-study-config--get-path" doesn't start with package's prefix "boa-ide".
54:0: error: "boa-study-config-current-context" doesn't start with package's prefix "boa-ide".
102:0: error: "boa-study-config-completion-at-point" doesn't start with package's prefix "boa-ide".
142:0: error: "boa-study-config-runnable-at-point" doesn't start with package's prefix "boa-ide".
159:0: error: "boa-study-config-run" doesn't start with package's prefix "boa-ide".
173:0: error: "boa-study-config-ffap-file" doesn't start with package's prefix "boa-ide".
191:0: error: "boa-study-config-ffap" doesn't start with package's prefix "boa-ide".
201:0: error: "boa-study-config-mode-map" doesn't start with package's prefix "boa-ide".
212:0: error: "boa-study-config-maybe-enable" doesn't start with package's prefix "boa-ide".
222:0: error: "boa-study-config-insinuate" doesn't start with package's prefix "boa-ide".
226:0: error: "boa-study-config-reparse" doesn't start with package's prefix "boa-ide".
230:0: error: "boa-study-config-mode" doesn't start with package's prefix "boa-ide".
```

It looks like you have a dependency on json-snatcher but note that since `Package-Requires outside the main file have no effect` (above)  you need to move all of your `Package-Requires` into the main file (boa-ide.el in this case).  This is preventing byte-compilation:

`boa-study-config.el` with byte-compile using Emacs 28.1:
```
boa-study-config.el:32:1: Error: Cannot open load file: No such file or directory, json-snatcher
```



`boa-study-config.el` with byte-compile using Emacs 28.1:
```
boa-study-config.el:32:1: Error: Cannot open load file: No such file or directory, json-snatcher
```

Reply:
These should not be fixed.